pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -150,6 +159,29 @@ protected void disableReplicatorRead() {
     protected void readMoreEntries() {
         int availablePermits = producerQueueSize - PENDING_MESSAGES_UPDATER.get(this);
 
+        // handle rate limit
```

Comment:
This could go in a separate private method.

Reply:
@merlimat , thanks for the comments,  updated